src/views/BrowseInventory/BrowseInventory.js

```diff
@@ -57,39 +66,39 @@ const BrowseInventory = () => {
   useEffect(() => {
     storeLastBrowse(search);
     storeLastBrowseOffset(pageConfig);
-  }, [search, pageConfig]);
+  }, [search, pageConfig, storeLastBrowse, storeLastBrowseOffset]);
 
   const [
     filters,
     searchQuery,
     applyFilters,
     applySearch,
     changeSearch,
-    resetFilters,
+    resetAll,
     changeSearchIndex,
     searchIndex,
+    clearFilters,
   ] = useLocationFilters(location, history, () => {
     setPageConfig(INIT_PAGE_CONFIG);
   });
 
   const withExtraFilters = useMemo(() => {
-    if (filters.query && INDEXES_WITH_CALL_NUMBER_TYPE_PARAM.includes(filters.qindex)) {
-      return {
-        ...filters,
-        callNumberType: filters.qindex,
-      };
-    }
+    const { qindex, query } = querystring.parse(search);
```

Comment:
That's because we need to reset filters because in some cases different browse options have different supported filters
I'll discuss it with Christine because that's a special case and requires more effort to handle.